kotoed-js/src/main/ts/profile/edit.tsx

```diff
@@ -200,6 +208,57 @@ export class ProfileComponent extends ComponentWithLocalErrors<ProfileComponentP
         </div>;
     };
 
+    renderDiffPreference = () => {
+        return <div className="form-group">
+            <label htmlFor="diff-mode-preference" className="col-sm-2 control-label">Preferred diff mode</label>
+            <div className="col-sm-10">
+                <div className="radio">
+                    <label htmlFor="input-diff-mode-course-base">
+                        <input
+                            id={`input-diff-mode-course-base`}
+                            type="radio"
+                            name="diff-mode"
+                            value="COURSE_BASE"
+                            checked={this.props.denizen.diffModePreference == "COURSE_BASE"}
+                            onChange={this.bindRadio("diffModePreference")}
+                            defaultChecked={this.state.denizen.emailNotifications}
+                        />
+                        Course base revision
+                    </label>
+                </div>
+                <div className="radio">
+                    <label htmlFor="input-diff-mode-previous-closed">
+                        <input
+                            id={`input-diff-mode-previous-closed`}
+                            type="radio"
+                            name="diff-mode"
+                            value="PREVIOUS_CLOSED"
+                            checked={this.props.denizen.diffModePreference == "PREVIOUS_CLOSED" ||
+                                !this.props.permissions.tags && this.props.denizen.diffModePreference == "PREVIOUS_CHECKED" }
```

Comment:
Looks like this is incorrect state but maybe it's ok to handle it in a such way

Reply:
Trying to handle a situation which probably will never happen. If user was a teacher and chosen `PREVIOUS_CHECKED` and then his teacher permission is revoked, the preference might not be changed. Here we are trying to handle it.